targets/TARGET_NUVOTON/TARGET_NUC472/device/StdDriver/nuc472_clk.c

```diff
@@ -75,6 +75,8 @@ void CLK_PowerDown(void)
     SCB->SCR = SCB_SCR_SLEEPDEEP_Msk;
     CLK->PWRCTL |= (CLK_PWRCTL_PDEN_Msk | CLK_PWRCTL_PDWKDLY_Msk );
     __WFI();
+    __DSB();
+    __ISB();
```

Comment:
@0xc0170 , according to the hard fault prompt & tracking, it's failed in branch instruction after wake-up. So add these barriers to ensure data & instruction ready before branch.  Cross checking of my test proved it could work as our expectation.